File: movie_api_prj/src/main/java/com/example/mpi/service/MovieService.java
package com.example.mpi.service;

import java.util.List;

import com.example.mpi.dto.MovieDto;

public interface MovieService {
	// 메인페이지 개봉예정작 리스트
	List<MovieDto> getMovieForMainUpComing(String adult);
	// 메인페이지 개봉작 최신 영화 리스트
	List<MovieDto> getMovieForMainLatest(String adult);
	// 메인페이지 평점순 영화 리스트
	List<MovieDto> getMovieForMainTop(String adult);
	// 메인페이지 인기순 영화 리스트
	List<MovieDto> getMovieForMainPopular(String adult);
	// 영화 디테일
	MovieDto getMovieDetail(String id);
}

File: movie_api_prj/frontend/src/services/movie.service.js
import { instance } from './url'

class MovieService {
  getMovieForMainUpComing(adult) {
    return instance.get(`/movie/main/upcoming/${adult}`)
  }

  getMovieForMainLatest(adult) {
    return instance.get(`/movie/main/latest/${adult}`)
  }

  getMovieForMainTop(adult) {
    return instance.get(`/movie/main/top/${adult}`)
  }

  getMovieForMainPopular(adult) {
    return instance.get(`/movie/main/popular/${adult}`)
  }

  getMovieDetail(id) {
    return instance.get(`/movie/detail/${id}`)
  }

  getMovieListAll(pageNo) {
    return instance.get(`/movie/list/${pageNo}`)
  }

  getMovieListAllForResult(pageNo, genre, sortBy) {
    return instance.get(`/movie/list/${pageNo}/${genre}/${sortBy}`)
  }

  getMovieListAllForResultTitle(pageNo, title) {
    return instance.get(`/movie/list/${pageNo}/${title}`)
  }
}

export default new MovieService()
